source/main/providers/link-provider.js

```diff
@@ -0,0 +1,266 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        LinkProvider class
+ * CVM-Role:        Service Provider
+ * Maintainer:      Julien Mirval
+ * License:         GNU GPL v3
+ *
+ * Description:     Handles everything link related that's going on in the app.
+ *
+ * END HEADER
+ */
+
+/**
+ * This class manages note's relations on the app. It reads the links on each
+ * start of the app and writes them after they have been changed.
+ */
+class LinkProvider {
+  /**
+   * Create the instance on program start and initially load the links.
+   * @param {FSALCache} cache a cache to store links
+   */
+  constructor () {
+    global.log.verbose('Link provider booting up ...')
+
+    this._links = []
+    // The global link database; it contains all links that are used in any of the
+    // files.
+    this._globalLinkDatabase = Object.create(null)
+
+    // Register a global helper for the link database
+    global.links = {
+      /**
+       * Adds an array of links to the database
+       * @param  {Array} linkArray An array containing the links to be added
```

Comment:
Aside: As a general suggestion about JSDoc comments, I believe it would be better if you could use generic types (e.g. `Array<String>`) instead of the non-generic version (e.g. `Array`). This would help others understand the code and (I guess) make the editor provide better hints and type-checking.

This is not really important as Zettlr is migrating to TypeScript anyways.